From 733b995f1eb0bd19cb16cbce7164730645505297 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gmail.com>
Date: Sun, 28 Aug 2011 21:26:06 -0400
Subject: [PATCH] Gdk-3.0 supports multiple targets

Gdk-3.0 supports multiple targets now, and the "target" variable no
longer works.

[ This is an 0.29.0 port of a patch created by Nirbheek Chauhan
<nirbheek@gentoo.org> for 0.28.0. ]

https://bugzilla.gnome.org/show_bug.cgi?id=657581
---
 configure.in |   16 +++++++++-------
 1 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/configure.in b/configure.in
index e89106d..a4ef0b7 100644
--- a/configure.in
+++ b/configure.in
@@ -378,13 +378,15 @@ esac
 # Search for the required modules.
 
 # We have a direct dependency on X11 on gdk-x11, see bug #613525
-AC_MSG_CHECKING([for GDK target])
-GDK_TARGET="$($PKG_CONFIG --variable target gdk-$GTK_API_VERSION)"
-AC_MSG_RESULT([$GDK_TARGET])
-case "$GDK_TARGET" in
-  x11) PLATFORM_PKGS="x11 cairo-xlib" ;;
-  *) PLATFORM_PKGS="" ;;
-esac
+AC_MSG_CHECKING([for GDK targets])
+GDK_TARGETS="$($PKG_CONFIG --variable targets gdk-$GTK_API_VERSION)"
+AC_MSG_RESULT([$GDK_TARGETS])
+PLATFORM_PKGS=
+for i in $GDK_TARGETS; do
+  if test "$i" = "x11"; then
+    PLATFORM_PKGS="$PLATFORM_PKGS x11 cairo-xlib"
+  fi
+done
 
 VTE_PKGS="glib-2.0 >= $GLIB_REQUIRED gobject-2.0 pango >= $PANGO_REQUIRED gtk+-$GTK_API_VERSION >= $GTK_REQUIRED gobject-2.0 gio-2.0 gio-unix-2.0 $PLATFORM_PKGS"
 PKG_CHECK_MODULES([VTE],[$VTE_PKGS])
-- 
1.7.6.1

